src/ResourceManager/ContainerRegistry/Commands.ContainerRegistry/Commands/UpdateAzureContainerRegistry.cs

```diff
@@ -90,16 +82,32 @@ public override void ExecuteCmdlet()
                     adminUserEnabled = EnableAdminUser || !DisableAdminUser;
                 }
 
-                string storageAccountResourceGroup = null;
+                string storageAccountId = null;
 
                 if (StorageAccountName != null)
                 {
-                    storageAccountResourceGroup = ResourceManagerClient.GetStorageAccountResourceGroup(StorageAccountName);
+                    storageAccountId = ResourceManagerClient.GetStorageAccountId(StorageAccountName);
+                }
+
+                if (MyInvocation.BoundParameters.ContainsKey("ResourceId") || !string.IsNullOrWhiteSpace(ResourceId))
+                {
+                    string resourceGroup, registryName, childResourceName;
+                    if(!ConversionUtilities.TryParseRegistryRelatedResourceId(ResourceId, out resourceGroup, out registryName, out childResourceName))
+                    {
+                        WriteInvalidResourceIdError(InvalidRegistryResourceIdErrorMessage);
+                        return;
+                    }
+
+                    ResourceGroupName = resourceGroup;
+                    Name = registryName;
                 }
 
-                var registry = RegistryClient.UpdateRegistry(
-                    ResourceGroupName, Name, adminUserEnabled, StorageAccountName, storageAccountResourceGroup, tags);
-                WriteObject(new PSContainerRegistry(registry));
+                if (ShouldProcess(Name, "Update Container Registry"))
```

Comment:
@Jinhuafei there is already a `ShouldProcess` call on line 74. I would remove that block in favor of this one.

Reply:
Fixed.